Main.java

```diff
@@ -5,61 +5,58 @@
 
 public class Main {
 
-    public static void main(String[] args){
+    public static void main(String[] args) {
 
-        if(args.length!=2)
-        {
+        if (args.length != 2) {
             System.out.println("File path and search word cannot be empty");
-            System.exit(0);
+            return;
         }
-        String path=args[0];
-        String search=args[1];
-        File file=new File(path);
+        String path = args[0];
+        String search = args[1];
+        File file = new File(path);
         System.out.println("Processing...");
-        doesFileExist(file);
-        wordCount(file,search);
+        validateFile(file);
+        if (!validateFile(file)) {
+            wordCount(file, search);
+        }
     }
 
-    private static void doesFileExist(File file)
-    {
-        if (!file.isFile())
-        {
-            System.out.println("File does not exist");
-        } else isFileFormatSupported(file);
-    }
+    private static boolean validateFile(File file) {
 
-    private static void isFileFormatSupported(File file)
-    {
-        if (!file.getName().endsWith("txt"))
-        {
-            System.out.println("File format not supported");
+        if (!file.exists()) {
+            System.out.println("File does not exist");
+            System.exit(0);
+        } else {
+            String fullName = file.getName();
+            if (!(fullName.endsWith(".txt") || fullName.endsWith(".json"))) {
+                System.out.println("File format not supported");
+                System.exit(0);
+            }
         }
+        return false;
     }
 
-    private static int wordCount(File file,String search){
+    private static void wordCount(File file, String search) {
 
-        String data = null;
-        String removeSpecialChar="";
+        String data = "";
         try {
             FileReader reader = new FileReader(file);
             BufferedReader buffer = new BufferedReader(reader);
             data = buffer.readLine();
         } catch (Exception e) {
             e.printStackTrace();
         }
-        if(data!=null) {
-                removeSpecialChar = data.replaceAll("\\p{Punct}", "");
-        }else {
-                System.out.println("Empty file");
-                System.exit(1);
-            }
-        StringTokenizer st = new StringTokenizer(removeSpecialChar);
+        if (data != null) {

```

Comment:
Data will never be null since data you are setting as data="" in line 41

Reply:
But when I passed an empty file, the code worked fine and it showed "File does not contain any data". So the else condition works, right?